play-frontend-hmrc-play-30/src/test/scala/uk/gov/hmrc/hmrcfrontend/views/components/HmrcFooterSpec.scala

```diff
@@ -17,6 +17,25 @@
 package uk.gov.hmrc.hmrcfrontend.views
 package components
 
+import play.api.i18n.Messages
+import play.api.mvc.MessagesRequest
+import play.api.test.FakeRequest
+import play.twirl.api.HtmlFormat
+import uk.gov.hmrc.helpers.MessagesSupport
 import uk.gov.hmrc.hmrcfrontend.views.html.components._
 
-class HmrcFooterSpec extends TemplateUnitSpec[Footer, HmrcFooter]("hmrcFooter")
+import scala.util.Try
+
+class HmrcFooterSpec extends TemplateUnitBaseSpec[Footer]("hmrcFooter") with MessagesSupport {
+
+  private val component = app.injector.instanceOf[HmrcFooter]
+
+  def render(templateParams: Footer): Try[HtmlFormat.Appendable] = {
+    val request = FakeRequest("GET", "/foo")
+      .withTransientLang(templateParams.language.code)
+
+    implicit val messages: Messages = new MessagesRequest(request, messagesApi).messages
```

Comment:
small one, but I think we could reduce this to
```suggestion
    implicit val messages = messagesApi.preferred(Seq(Lang(templateParams.language.code)))
```
MessagesSupport trait could also be changed to `messages = mesagesApi.preferred(FakeRequest())` or use the seq in the defaultLangs in that trait rather than a fake request, but 🤷‍♂️

Reply:
Nice - works a treat 🙌